src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1799,31 +1799,38 @@ else if (con.getTrustManagerClass() != null) {
                                 + tmfDefaultAlgorithm + "\n") : "")
                         + ((null != ksProvider) ? ("KeyStore provider info: " + ksProvider.getInfo() + "\n") : "")
                         + "java.ext.dirs: " + System.getProperty("java.ext.dirs"));
+            // Retrieve the localized error message if possible.
+            String errMsg = (e.getLocalizedMessage() != null) ? e.getLocalizedMessage() : e.getMessage();
+            /*
+             * Retrieve the error message of the cause too because actual error message can be wrapped into a different
+             * message when re-thrown from underlying InputStream.
+             */
+            String causeErrMsg = null;
+            if (e.getCause() != null) {
+                causeErrMsg = (e.getCause().getLocalizedMessage() != null) ? e.getCause().getLocalizedMessage()
+                                                                           : e.getCause().getMessage();
+            }
```

Comment:
I like conditionals it looks cleaner but this means getLocalizedMessage() and getCause() are called multiple times

Reply:
These are just getters, it wouldn't really make much difference, but I still updated the code in case if there are changes to these methods in the future.